beacon-chain/core/blocks/block_operations.go

```diff
@@ -31,13 +31,44 @@ import (
 // transition processing.
 //
 // WIP - this is stubbed out until BLS is integrated into Prysm.
-func VerifyProposerSignature(
-	ctx context.Context,
-	_ *pb.BeaconBlock,
-) error {
+func VerifyProposerSignature(ctx context.Context, beaconState *pb.BeaconState, block *pb.BeaconBlock) error {
 	ctx, span := trace.StartSpan(ctx, "beacon-chain.ChainService.state.ProcessBlock.VerifyProposerSignature")
 	defer span.End()
 
+	proposerIndex, err := helpers.BeaconProposerIndex(beaconState, block.Slot)
+	if err != nil {
+		return fmt.Errorf("could not get beacon proposer index at slot %v: %v", block.Slot, err)
+	}
+
+	proposerPubkey, err := bls.PublicKeyFromBytes(beaconState.ValidatorRegistry[proposerIndex].Pubkey)
+	if err != nil {
+		return fmt.Errorf("could not get proposer public key: %v", err)
+	}
+
+	blockRootHash, err := hashutil.HashBeaconBlock(block)
+	if err != nil {
+		return fmt.Errorf("could not hash beacon block: %v", err)
+	}
+
+	proposalHash, err := hashutil.HashProposal(&pb.Proposal{
+		Slot:            block.Slot,
+		Shard:           params.BeaconConfig().BeaconChainShardNumber,
+		BlockRootHash32: blockRootHash[:],
+	})
+	if err != nil {
+		return fmt.Errorf("could not hash attestation data: %v", err)
+	}
+
+	sig, err := bls.SignatureFromBytes(block.Signature)
+	currentEpoch := helpers.CurrentEpoch(beaconState)
+
+	domain := forkutils.DomainVersion(beaconState.Fork, currentEpoch, params.BeaconConfig().DomainProposal)
+
+	log.WithFields(logrus.Fields{}).Info("Verifying proposal")
```

Comment:
Please don't `Info` this. Let's use `debug` and add more context to the log `Verifying proposal hash %#x etc`